command-expansion-server/src/lib/codegen/ActivityTypescriptCodegen.ts

```diff
@@ -53,8 +53,12 @@ function convertSchemaType(schema: Schema): string {
           .join('\n'),
       );
     case SchemaTypes.Variant:
-      if (schema.variants.length === 1 && schema.variants[0]?.key === 'VOID') {
-        return 'null';
+      if (
+        schema.variants.length === 1 &&
+        (schema.variants[0]?.key === 'VOID' ||
+          (schema.variants[0]?.key === 'UNIT' && schema.variants[0]?.label === 'UNIT'))
```

Comment:
Would you see any downsides to removing the specialization to the particular keys `VOID` or `UNIT`? The latter in particular is automatically derived from the name of a Java identifier, and it makes me uncomfortable to imagine that refactoring a name might cause a user-visible change (and a bug, at that).

Alternatively, should we consider adding a new type of `ValueSchema` that explicitly denotes the lack of a useful value? That would make this situation easily detectable without causing all one-variant enums to be treated as trivial.

Reply:
*Alternatively* alternatively, perhaps we represent UNIT as an empty struct instead of a one-variant enum? Then we have a very close match indeed to the representation here as `{}`.

Maybe that would mean we don't have to change anything at all in the typescript, because I imagine it *already* turns empty maps into `{}`.

I think I like this the best. Thoughts?

Reply:
I was also wondering why we need to check length and key. Can we have a schema with a `key` and `label` of `UNIT` but not a `length` of 1? If not, then I don't think it ought to be `&&`'d with the length.

Reply:
@Mythicaeda, in principle, nothing's stopping a modeler from writing an `enum Foo { UNIT; AAA; BBB }`, where it has three variants but one of them happens to be called "UNIT". (Another reason I don't like the name-sniffing.)

Reply:
For demonstration, I pushed a branch [`feature/unit-serialization`](https://github.com/NASA-AMMOS/aerie/commit/0ebb9928f6ef5eb983c513bbc3e8200b148e7200). This should override the representation of `Unit.UNIT` so that it is represented as `{}` from the very beginning, and so ostensibly should not require any changes downstream here. Please do share your thoughts 😃

Reply:
Does `{}` have a length of 0 or 1?

Reply:
Zero. `Unit` is currently serialized as a list of variants (of length 1); I'm proposing serializing it as a struct with no fields (so 0).

But the code branch this comment is on only applies to variants, not structs, so this would push `Unit` into a different branch entirely. The relevant preexisting case is copied here:
```ts
    case SchemaTypes.Struct:
      return objectDeclaration(
        Object.entries(schema.items)
          .map(([key, value]) => `${key}: ${convertSchemaType(value)};`)
          .join('\n'),
      );
```
Since `items` will be empty, this should produce an empty object `{}` without any further effort on our part.

Reply:
Hi Jonathan,

Thank you for all the feedback. I do like your idea of a value mapper that changes it upstream. I wasn't sure how to hook up the value mapper and your branch helps a lot. It is cleaner than my approach. Let me play around with the code as I want to learn a little bit more about the valuemappers. I haven't really touched that part of AERIE.